mantle/kola/harness.go

```diff
@@ -588,24 +587,48 @@ RequiredBy=multi-user.target
 			mach := c.Machines()[0]
 			plog.Debugf("Running kolet")
 
+			var previousRebootState string
+			var stdout []byte
 			var stderr []byte
 			var err error
 			for {
 				plog.Debug("Starting kolet run-test-unit")
-				_, stderr, err = mach.SSH(fmt.Sprintf("sudo ./kolet run-test-unit %s", shellquote.Join(unitname)))
+				if previousRebootState != "" {
+					plog.Debugf("Setting AUTOPKGTEST_REBOOT_MARK=%s", previousRebootState)
+					c.MustSSHf(mach, "echo AUTOPKGTEST_REBOOT_MARK=%s | sudo tee /run/kola-runext-env", previousRebootState)
+					previousRebootState = ""
+				}
+				stdout, stderr, err = mach.SSH(fmt.Sprintf("sudo ./kolet run-test-unit %s", shellquote.Join(unitname)))
+				expectingReboot := false
 				if exit, ok := err.(*ssh.ExitError); ok {
 					plog.Debug("Caught ssh.ExitError")
-					// In the future I'd like to better support having the host reboot itself and
-					// we just detect it.
+					// If we got SIGTERM, then we assume the unit (or our login) was killed by a reboot.
+					// Verify that on stdout
 					if exit.Signal() == "TERM" {
-						plog.Debug("Caught SIGTERM from kolet run-test-unit, rebooting machine")
+						plog.Debug("Caught SIGTERM from kolet run-test-unit")
+						expectingReboot = true
+						err = nil
+					}
+				}
+				if err == nil {
+					stdout := strings.TrimSpace(string(stdout))
+					if stdout == "reboot" {
+						var kolaRebootStdout []byte
+						kolaRebootStdout, _, err = mach.SSH("if test -f /run/kola-reboot; then cat /run/kola-reboot; fi")
```

Comment:
Ahh I was trying to figure out where the code that read the mark was by Ctrl+F'ing for `kolaRebootStamp`. Can we use that const here too?

Reply:
Thoughts on this one?

Reply:
Centralized the stamp file naming.